nixos/modules/system/boot/luksroot.nix

```diff
@@ -432,11 +432,39 @@ let
           echo "Please move your mouse to create needed randomness."
         ''}
           echo "Waiting for your FIDO2 device..."
-          fido2luks open ${dev.device} ${dev.name} ${dev.fido2.credential} --await-dev ${toString dev.fido2.gracePeriod} --salt string:$passphrase
+          ${optionalString dev.fido2.askForPin ''
+            echo -n "Enter PIN for authenticator: "
+            pin=
+            while true; do
+                if [ -e /crypt-ramfs/passphrase ]; then
+                    echo "reused"
+                    pin=$(cat /crypt-ramfs/passphrase)
+                    break
+                else
+                    # Try reading it from /dev/console with a timeout
+                    IFS= read -t 1 -r pin
+                    if [ -n "$pin" ]; then
+                       ${if luks.reusePassphrases then ''
+                         # Remember it for the next device
+                         echo -n "$pin" > /crypt-ramfs/passphrase
```

Comment:
I think the code only works if reusePassphrases is set to true. If reusePassphrases is set to false, /crypt-ramfs/passphrase is never written and fido2luks --pin-source will point to a not existing file.

Reply:
Good catch. I pushed up a change that fixes this.